### PR TITLE
updating helper for css variables

### PR DESCRIPTION
**Pull request by iruzevic**
adding options to convert css variable output to color, select-variable, boolean-variable and default. also option to add any css variable no mater the manifest

### CODE REVIEW

File: src/Helpers/Components.php [hunk 1]
```diff
@@ -382,19 +382,47 @@ public static function outputCssVariables(array $attributes, array $manifest, st
...
 			}
 
-			if (isset($manifest['attributes'][$key]['color'])) {
+			// Output color variable from the global variables.
+			if ($manifest['attributes'][$key]['variable'] === 'color') {
 				$value = "var(--global-colors-{$value})";
 			}
 
+			// Output select variable from the options array but dont use value key. It will use variable key.
+			if ($manifest['attributes'][$key]['variable'] === 'select-variable' && isset($manifest['options'][$key])) {
```
Comment (dingo-d) [inline src/Helpers/Components.php:391]: I'd put the isset key check first, then the specific value second. Just think it's more readable and logical (checking if it's set first, then the value)

File: src/Helpers/Components.php [hunk 2]
```diff
@@ -382,19 +382,47 @@ public static function outputCssVariables(array $attributes, array $manifest, st
...
+					function ($item) use ($attributes, $key) {
+						return $item['value'] === $attributes[$key];
+					}
+				))[0]['variable'] ?? null;
+
+				$value = $selectVariable === null ? $attributes[$key] : $selectVariable;
+			}
+
+			// Output boolean variable from the options array key. First key is false value, second is true value.
+			if ($manifest['attributes'][$key]['variable'] === 'boolean-variable' && isset($manifest['options'][$key]) && count($manifest['options'][$key]) === 2) {
```
Comment (dingo-d) [inline src/Helpers/Components.php:403]: Same here. Isset first, then other checks.

File: src/Helpers/Components.php [hunk 3]
```diff
@@ -382,19 +382,47 @@ public static function outputCssVariables(array $attributes, array $manifest, st
...
+				$value = $manifest['options'][$key][(int)$attributes[$key]];
+			}
+
+			// Return correct boolean type as string.
+			if (gettype($value) === 'boolean') {
+				$value = $value ? 'true' : 'false';
```
Comment (dingo-d) [inline src/Helpers/Components.php:409]: I think you could just typecast to a string `(string)$value`.

File: src/Helpers/Components.php [hunk 4]
```diff
@@ -382,19 +382,47 @@ public static function outputCssVariables(array $attributes, array $manifest, st
...
+			}
+
 			$key = self::camelToKebabCase($key);
 
 			$output .= "--{$key}: {$value};\n";
 		}
 
+
+		// Output manual output from the array ov variables.
```
Comment (dingo-d) [inline src/Helpers/Components.php:418]:
```suggestion
		// Output manual output from the array of variables.
```

**Review by dingo-d — CHANGES_REQUESTED** [review]
Left some small comments. Awesome that you have tests so you can check if anything breaks after changing the code 😁😁

File: tests/Helpers/ComponentHelpersTest.php [hunk 5]
```diff
@@ -359,12 +359,11 @@ function ($argument) {
 /**
  * Components::outputCssVariables tests
  */
-test('Asserts that outputCssVariables returns the correct css variables output', function () {
+test('Asserts that outputCssVariables returns the correct css variables output for default type', function () {
```
Comment (goranalkovic-infinum) [inline tests/Helpers/ComponentHelpersTest.php:362]:
```suggestion
test('Asserts that outputCssVariables returns the correct CSS variables output for default type', function () {
```

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]
Left one suggestion, @dingo-d said the rest 😄  Looks awesome!

File: src/Helpers/Components.php (hunk 3 above)
Comment (volfkarlo) [inline src/Helpers/Components.php:409]: I'm not fully sure how would do you handle true or false in CSS variable world. 
Boolean in CSS vars, for now, is mostly resolved by 1 or 0 multiplied or by a value that you want to provide, or strings that you pass for example
```
--displaying-variable: block;
--displaying-variable: none;
display: var(--displaying-variable);
```

**Review by volfkarlo — CHANGES_REQUESTED** [review]

File: src/Helpers/Components.php (hunk 3 above)
Comment (iruzevic) [reply src/Helpers/Components.php:409]: I have tried it and it returns 0 and 1 as string.

**Review by goranalkovic-infinum — APPROVED** [review]

**Comment by iruzevic** [comment]
Implemented all corrections.

**Review by volfkarlo — APPROVED** [review]

**Pull request merged**